src/Template/Slot.php

```diff
@@ -21,14 +21,13 @@ class Slot
 	/**
 	 * Keeps track of the slot state
 	 */
-	public bool $open = false;
+	protected bool $open = false;
 
 	/**
 	 * Creates a new slot for the given snippet
 	 */
 	public function __construct(
-		public Snippet $snippet,
-		public string $name,
+		protected string $name,
 		public string|null $content = null
```

Comment:
Why can't both be public? Doesn't make most sense to me to have one protected, one public.

Reply:
`$content` needs to be "internally public" because it is set from the `Snippet` class. `$name` on the other hand is supposed to be readonly.

The reason I made the props `protected` is because we use protected props basically all around the core. Only very recent PRs use `public` props just to make testing easier. But public props at the same time reduce the real-world reliability as it's very easy to override them by accident. Also they would all end up in auto-generated docs (e.g. IDE code suggestions), which would make it seem like we actually *intend* the props to be set from the outside. Access from tests should IMO never be more important than the intended semantics.

Reply:
Ok, I agree with that.

Just wondering about property order then. In this specific case, it would seem a bit odd that content comes before name, to me at least.

Reply:
This is a thing I was wondering about in other places with the new PHP 8 style. Using the property shortcuts seems super useful at first, but we also lose a bit of control over here. But I think this is something we have to find out bit by bit with more use cases. The property order in the constructor should first be determined by logic/human aspects. If it does not fit the desired property order in the class, we should maybe abandon the shortcuts in such cases.

Reply:
I agree. I'd move the properties to "real" properties in this case. This also allows to easily add an `@internal` hint without blowing up the constructor arguments.